src/Servers/IIS/IIS/src/Core/IISHttpContext.FeatureCollection.cs

```diff
@@ -290,23 +307,23 @@ ClaimsPrincipal IHttpAuthenticationFeature.User
                 // Synchronize access to native methods that might run in parallel with IO loops
                 lock (_contextLock)
                 {
-                    NativeMethods.HttpSetServerVariable(_requestNativeHandle, variableName, value);
+                    NativeMethods.HttpSetServerVariable(_requestNativeHandle, variableName, value!); // TODO: What happens if you set a null server variable value?
```

Comment:
Let's guard against calling this API with a null value.

Reply:
How do you remove a server variable? Set it to `string.Empty`?

Reply:
That would be my guess. Docs don't mention what happens if you pass in null here: https://docs.microsoft.com/en-us/iis/web-development-reference/native-code-api-reference/ihttpcontext-setservervariable-method

It's also not common to set server variables in the first place.